relentless/optimize/optimizer.py

```diff
@@ -0,0 +1,23 @@
+__all__ = ['Optimizer','SteepestDescent']
+
+import abc
+
+from .objective import ObjectiveFunction
+
+class Optimizer(abc.ABC):
+    def __init__(self, obj):
+        self.objective = obj
+
+    @abc.abstractmethod
+    def run(self):
+        pass
+
+class SteepestDescent(Optimizer):
+    def __init__(self, obj, alpha):
+        super().__init__(obj)
+        self.alpha = alpha
+
+    def run(self):
+        res = self.objective.run()
+        for x in res.gradient:
+            x.value += self.alpha*res.gradient[x]
```

Comment:
This will get fleshed out more later, since it's missing constraints and stopping criteria right now. It also needs to iterate to convergence. I think also the sign is wrong and should be `-=`?